integtest/spec/helper/dsl/file_contexts.rb

```diff
@@ -0,0 +1,65 @@
+# frozen_string_literal: true
+
+module Dsl
+  ##
+  # Methods to create contexts for asserting things about files and
+  # their contents.
+  module FileContexts
+    ##
+    # Create a context to assert things about a file. By default it just
+    # asserts that the file was created but if you pass a block you can add
+    # assertions on `contents`.
+    def file_context(name, file_name = name, &block)
+      context "for #{name}" do
+        include_context 'Dsl_file', file_name
+
+        # Yield to the block to add more tests.
+        class_exec(&block)
+      end
+    end
+    shared_context 'Dsl_file' do |file_name|
+      let(:file) do
+        dest_file(file_name)
+      end
+      let(:contents) do
+        return unless File.exist? file
```

Comment:
It looks strange to me to have a `return` in a `let` block. Maybe you can just write the line in a `if File.exist? file`

Reply:
It does feel weird? I'm not really sure the `if File.exist?` block is super clear either.  I think for me coming from Java the `return unless` is more clear but I can't speak to idiomatic ruby.

Reply:
I guess I think of `let` blocks not quite as methods, more like procs. But this is not a critical thing and if you prefer using `return`, that's no problem.

Reply:
Thanks!

Let me see if I can resolve the merge conflicts here.....